BUG: Fix apply_along_axis() for when func1d() returns a non-ndarray

bennyrowland opened the pull request:
Fixes issue in apply_along_axis() where func1d() returns a non ndarray

bennyrowland committed:
BUG: Closes issue #8419

Fixes issue in apply_along_axis() where func1d() returns a non ndarray

charris commented on numpy/lib/tests/test_shape_base.py:
```diff
@@ -58,6 +58,10 @@ class MinimalSubclass(np.ndarray):
         assert_array_equal(res, np.array([6, 6, 6]).view(MinimalSubclass))
 
+    def test_tuple_func1d(self):
+        res = np.apply_along_axis((lambda x: (x[1], x[0])), 1, np.array([[1, 2], [3, 4]]))
```
Line is too long, please keep under 80 characters.

charris commented:
Fix the long line, otherwise LGTM. The commit message summary line is pretty uninformative, note the edited title, and`Closes #8419` should be in the message body.

bennyrowland committed:
BUG: Fix apply_along_axis() when func1d() returns a non-ndarray

Closes issue #8419. Fixes issue in apply_along_axis() where
func1d() returns a non ndarray by calling asanyarray() on
result. This commit fixes a too long line in the test case.

charris commented:
Thanks @bennyrowland. I did a squash merge here. You can do the squashing yourself next time by using `git rebase -i`, which is well worth the time to become familiar with.
